### ISSUE TITLE
objlist from frame

### ISSUE THREAD

**Issue by profxj** [desihub/desispec#690]
On developing QA and intending to separate the plotting of 
metrics by granular object type (e.g. ELG, LRG, QSO), I am finding
myself confused on the current convention.  Here is example
output for the 18.7 mini-test:

In [1]:     from desispec.io import read_frame
In [2]: frame = read_frame('frame-b0-00000021.fits')
In [3]: objlist = set(frame.fibermap["OBJTYPE"])
In [4]: objlist
Out[4]: {'SCIENCE', 'SKY', 'STD'}

i.e. no granularity in the fibermap.

But the QA figure that @rstaten shared with me the other day (attached)
suggests the ELG, LRG, typing might can come from the fibermap['OBJTYPE'] items.
That is, looking at line 2057 of qa_quicklook.py
I can see that the Calculate_SNR class is grabbing objlist in the
same manner as above.  So I find myself confused.

Anyhow, I am mainly seeking guidance from @sbailey on the going-forward approach
to grabbing granular object types.  I suspect the answer is to use the
object flags, but my belief is that the QL QA is not yet adopting this approach.

![ql-snr-r0-00003577](https://user-images.githubusercontent.com/3998093/44670516-bc7a6200-a9d7-11e8-8cfb-86f423d74ba9.png)

**Comment by sbailey** [comment]
Indeed, this is a topic in transition.  OBJTYPE will no longer be used for distinguishing between different target types, e.g. because a single target could be both a QSO and an ELG target, and a standard star could also be a MWS science target.  The 18.7 minitest notebook is not in the final state either.  I'm currently testing a branch that just has 3 object types:
* TGT : a real object with expected flux, e.g. a science target or a standard star
* SKY : sky fiber, can be used for sky subtraction, not expecting object flux
* BAD : don't use this fiber for anything (maybe broken, maybe on a BADSKY location near a bright object, may be a stuck positioner).

To get the info you need for QA per target type, use the DESI_TARGET bits instead, e.g.
```python
from desitarget.targetmask import desi_mask
isELG = (fibermap['DESI_TARGET'] & desi_mask.mask('ELG')) != 0
```

When I have my desisim+desispec branches working with the new fibermap format, I'll likely need help with adapting QA and QL.  This is my top priority coding activity when I'm not juggling everything else.
